core/src/dpdk/port.rs

```diff
@@ -136,30 +135,25 @@ impl PortQueue {
         #[cfg(feature = "metrics")]
         self.received.as_ref().unwrap().record(len as u64);
 
-        let mbufs = unsafe {
-            // does a no-copy conversion to avoid extra allocation.
-            Vec::from_raw_parts(ptrs.as_mut_ptr() as *mut Mbuf, len as usize, RX_BURST_MAX)
-        };
-
         #[cfg(feature = "pcap-dump")]
-        pcap::append_and_write(self.port_id, CoreId::current(), "rx", &mbufs);
+        pcap::append_and_write(self.port_id, CoreId::current(), "rx", &ptrs);
 
-        mem::forget(ptrs);
-        mbufs
+        unsafe {
+            ptrs.set_len(len as usize);
+            ptrs.into_iter()
```

Comment:
this pattern may be enough to make it it's own more reusable/top-level fn? seems so?

Reply:
cuz these are unsafe calls, I feel it's better to inline them here. it's only repeated a couple (?) times. doesn't seem to be worth it to abstract that into another fn.